lib/streamlit/elements/exception.py

```diff
@@ -131,10 +147,40 @@ def marshall(exception_proto: ExceptionProto, exception: BaseException) -> None:
         )
 
     if is_uncaught_app_exception:
-        uae = cast(UncaughtAppException, exception)
-        exception_proto.message = _GENERIC_UNCAUGHT_EXCEPTION_TEXT
-        type_str = str(type(uae.exc))
-        exception_proto.type = type_str.replace("<class '", "").replace("'>", "")
+        show_error_details = config.get_option("client.showErrorDetails")
+
+        # Options for show error details config.
```

Comment:
nit: I would change it to "Options for 'showErrorDetails' config" to make it more obvious that this is the actual name of the config